skyportal/handlers/api/user_obj_list.py

```diff
@@ -0,0 +1,366 @@
+                    required: true
+                  obj_id:
+                    type: string
+                    required: true
+                  list_name:
+                    type: string
+                    required: true
+                    description: |
+                        Listing name for this item, e.g., "favorites".
+                        Multiple objects can be saved by the same user to different
+                        lists, where the list names are user-defined.
+                        List name must be a non-empty string starting with an
+                        alphanumeric character or underscore.
+                        (it must match the regex: /^\\w+/)
+
+        responses:
+          200:
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: '#/components/schemas/Success'
+                    - type: object
+                      properties:
+                        data:
+                          type: object
+                          properties:
+                            listing_id:
+                              type: integer
+                              description: New listing ID
+
+        """
+
+        return self.add_listing(error_if_exists=True)
+
+    @auth_or_token
+    def put(self):
+        """
+        ---
+        description: Add a listing, if it exists, ignore this call
+        requestBody:
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  user_id:
+                    type: integer
+                    required: true
+                  obj_id:
+                    type: string
+                    required: true
+                  list_name:
+                    type: string
+                    required: true
+                    description: |
+                        Listing name for this item, e.g., "favorites".
+                        Multiple objects can be saved by the same user to different
+                        lists, where the list names are user-defined.
+                        List name must be a non-empty string starting with an
+                        alphanumeric character or underscore.
+                        (it must match the regex: /^\\w+/)
+
+        responses:
+          200:
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: '#/components/schemas/Success'
+                    - type: object
+                      properties:
+                        data:
+                          type: object
+                          properties:
+                            listing_id:
+                              type: integer
+                              description: New or existing listing ID
+
+        """
+
+        return self.add_listing(error_if_exists=False)
```

Comment:
The post will be very useful for, e.g., front end command to add a source to favorites. 

If we don't have put, we will have to ask if the source already is on the list each and every time. 

I think if it is useful enough for that, it may be useful enough for other cases.

Reply:
Why do you have to ask? Say I have some listing, (1, ZTF20abcd, 'favorites') that already exists in the DB. I POST it again and the API tells me it's already there and does nothing to the database. Then I PUT it and the PUT does nothing to the database but returns a 200. Both of these leave the database in identical condition, but the POST gives me more info (that there was already an entry there). Not sure what I gain by having both.